src/main/java/org/opensearch/neuralsearch/search/query/TopDocsMerger.java

```diff
@@ -109,6 +108,10 @@ private ScoreDoc[] getMergedScoreDocs(ScoreDoc[] source, ScoreDoc[] newScoreDocs
         // doc_id | magic_number_2
         // ...
         // doc_id | magic_number_1
-        return scoreDocsMerger.merge(source, newScoreDocs, SCORE_DOC_BY_SCORE_COMPARATOR, false);
+        return scoreDocsMerger.merge(source, newScoreDocs, comparator(), sortAndFormats != null);
```

Comment:
encapsulate check for sorting enablement into a method, same check can be used in other places in this class